### PR TITLE
Add validation for deleted query graph nodes

### LINKED ISSUE #116: Query validation bug with deleted nodes
I can delete a node, have a floating edge, and still be able to submit. The edge panel also still has the id of the deleted node in its source or target.

### PR DESCRIPTION
**Pull request by maximusunc**
Closes #116

### CODE REVIEW

File: frontend/src/pages/question/questionBuilder/useNewQuestionPanel.js [hunk 1]
```diff
@@ -125,6 +125,10 @@ export default function useNewQuestionPanel() {
    * @param {string} id unique id of node or edge. i.e. n0, n1, e0...
    */
   function openPanel(type, id) {
+    if (type === 'node' && query_graph.nodes[id] && query_graph.nodes[id].deleted) {
```
Comment (alongreyber) [inline frontend/src/pages/question/questionBuilder/useNewQuestionPanel.js:128]: I'm not sure why but I am still able to open the NodePanel for a deleted node. I created a query graph with Disease -> Gene, deleted gene, and was still able to open gene.

Comment (alongreyber) [reply frontend/src/pages/question/questionBuilder/useNewQuestionPanel.js:128]: It looks like query_graph is empty for some reason when the panel is opened.

**Review by alongreyber — CHANGES_REQUESTED** [review]

File: frontend/src/pages/question/questionBuilder/useNewQuestionPanel.js (hunk 1 above)
Comment (maximusunc) [reply frontend/src/pages/question/questionBuilder/useNewQuestionPanel.js:128]: Weird, ok. I'll look into this some more.

**Review by alongreyber — CHANGES_REQUESTED** [review]
This now breaks the question builder entirely for me (blank page + exception):

https://user-images.githubusercontent.com/1634866/103660845-11c9c500-4f3c-11eb-9418-058bcf23e1f4.mov

**Review by alongreyber — APPROVED** [review]

**Pull request merged**